xds/pkg/server/server.go

```diff
@@ -33,40 +38,99 @@ const (
 	grpcMaxConcurrentStreams = 1000000
 )
 
+// HealthCheck implements the health check logic exposed by the server.
+// It is implemented by internal components of the server and returns
+// an error if the component is unhealthy.
+type HealthCheck func(ctx context.Context) error
+
 // Server is a wrapper around go-control-plane's xds server.
 type Server struct {
 	logger *log.Logger
-	// port is the port the server runs on.
-	port int16
+	// xdsPort is the port the XDS server runs on.
+	xdsPort int16
 	// snapshotCache is passed updated by us and read by go-control-plane
 	// to provided proxies with config from.
 	snapshotCache cache.SnapshotCache
 	// nodeIDCh is nilable. It is used as a callback mechanism from go-control-plane's server to
 	// let us know when a new proxy has connected. The proxy's ID is passed on the channel.
 	nodeIDCh chan<- string
+	// healthChecks contains the server's health check implementations.
+	healthChecks []HealthCheck
+	// adminPort is the port the admin server runs on.
+	adminPort int16
 }
 
 // NewServer returns a new Server
 func New(
 	logger *log.Logger,
-	port int16,
+	xdsPort int16,
 	snapshotCache cache.SnapshotCache,
 	nodeIDCh chan<- string,
+	healthChecks []HealthCheck,
+	adminPort int16,
 ) *Server {
 	return &Server{
 		logger: logger.WithFields(log.Fields{
 			"component": "server",
 		}).Logger,
-		port:          port,
+		xdsPort:       xdsPort,
 		snapshotCache: snapshotCache,
 		nodeIDCh:      nodeIDCh,
+		healthChecks:  healthChecks,
+		adminPort:     adminPort,
 	}
 }
 
+func (s *Server) startAdminServer(ctx context.Context, address string) {
```

Comment:
```suggestion
func (s *Server) Run(ctx context.Context) {
```

Move address into `New`, and then this object matches the New/Run pattern as everywhere else.

Reply:
updated to create address within the function, this is a helper function that is already being called from Server.Run